Exiled.API/Extensions/ItemExtensions.cs

```diff
@@ -28,22 +28,15 @@ public static class ItemExtensions
         /// </summary>
         /// <param name="item">The item to be checked.</param>
         /// <returns>Returns whether the <see cref="ItemType"/> is an ammo or not.</returns>
-        public static bool IsAmmo(this ItemType item) => item is ItemType.Ammo9x19 or ItemType.Ammo12gauge or ItemType.Ammo44cal or ItemType.Ammo556x45 or ItemType.Ammo762x39;
+        public static bool IsAmmo(this ItemType item) => item.GetAmmoType() != AmmoType.None;
 
         /// <summary>
         /// Check if an <see cref="ItemType">item</see> is a weapon.
         /// </summary>
         /// <param name="type">The item to be checked.</param>
         /// <param name="checkMicro">Indicates whether the MicroHID item should be taken into account or not.</param>
         /// <returns>Returns whether the <see cref="ItemType"/> is a weapon or not.</returns>
-        public static bool IsWeapon(this ItemType type, bool checkMicro = true) => type switch
-        {
-            ItemType.GunCrossvec or ItemType.GunCom45 or ItemType.GunLogicer or ItemType.GunRevolver or ItemType.GunShotgun or ItemType.GunAK
-                or ItemType.GunCOM15 or ItemType.GunCOM18 or ItemType.GunE11SR or ItemType.GunFSP9
-                or ItemType.ParticleDisruptor => true,
-            ItemType.MicroHID when checkMicro => true,
-            _ => false,
-        };
+        public static bool IsWeapon(this ItemType type, bool checkMicro = true) => type.GetFirearmType() != FirearmType.None || (checkMicro && type == ItemType.MicroHID);
```

Comment:
type.GetFirearmType() is not FirearmType.None || (checkMicro && type is ItemType.MicroHID);